File: presentation/src/main/java/com/github/anrimian/musicplayer/ui/editor/common/DeleteErrorHandler.java
package com.github.anrimian.musicplayer.ui.editor.common;

import android.os.Build;

import androidx.fragment.app.FragmentManager;

import com.github.anrimian.musicplayer.di.Components;
import com.github.anrimian.musicplayer.di.app.AppComponent;

public class DeleteErrorHandler extends ErrorHandler {

    public DeleteErrorHandler(FragmentManager fm,
                              Runnable onPermissionGranted,
                              Runnable onPermissionDenied) {
        super(fm, onPermissionGranted, () -> {
            AppComponent appComponent = Components.getAppComponent();
            appComponent.storageFilesDataSource().clearDeleteData();

            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R
                    && !appComponent.librarySettingsInteractor().isAppConfirmDeleteDialogEnabled()) {
                return;
            }
            onPermissionDenied.run();
        });
    }
}

File: app/src/main/java/com/github/anrimian/musicplayer/ui/editor/common/DeleteErrorHandler.kt
package com.github.anrimian.musicplayer.ui.editor.common

import android.os.Build
import androidx.activity.result.ActivityResultCaller
import com.github.anrimian.musicplayer.di.Components

class DeleteErrorHandler(
    activityResultCaller: ActivityResultCaller,
    onPermissionGranted: () -> Unit,
    onPermissionDenied: () -> Unit
) : ErrorHandler(activityResultCaller, onPermissionGranted, {
    val appComponent = Components.getAppComponent()
    appComponent.storageFilesDataSource().clearDeleteData()
    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.R
        || appComponent.librarySettingsInteractor().isAppConfirmDeleteDialogEnabled()) {
        onPermissionDenied()
    }
})